### PR TITLE
Include FlowIds to all requests that accept X-Flow-Id header

### PR DESCRIPTION
**Pull request by mtesseract**
Add FlowId to all requests that expect a X-Flow-Id header as documented at
https://zalando.github.io/nakadi/manual.html
New function: setFlowId.

### CODE REVIEW

**Comment by mtesseract** [comment]
Would you like to review, @etorreborre ?
The FlowId handling was not correct.
I have included an X-Flow-Id header according to https://zalando.github.io/nakadi/manual.html.

File: src/Network/Nakadi/EventTypes/EventType.hs [hunk 1]
```diff
@@ -44,18 +46,23 @@ eventTypeUpdate ::
...
-eventTypeUpdate eventTypeName eventType =
-  httpJsonNoBody ok200 []
-  (setRequestMethod "PUT"
-   . setRequestPath (path eventTypeName)
-   . setRequestBodyJSON eventType)
+eventTypeUpdate eventTypeName eventType = do
+  config <- nakadiAsk
+  httpJsonNoBody ok200 [] $
+    (setRequestMethod "PUT"
+     . setRequestPath (path eventTypeName)
```
Comment (etorreborre) [inline src/Network/Nakadi/EventTypes/EventType.hs]: This is a (very) minor thing but the code is not consistent w.r.t the order of calls `setRequestPath` and `includeFlowId`.

File: tests/Network/Nakadi/Tests/Common.hs [hunk 2]
```diff
@@ -104,6 +104,9 @@ recreateEvent eventTypeName eventType = do
   eventTypeCreate eventType
 
 delayedPublish :: (MonadNakadi b m, MonadIO m, ToJSON a) => Maybe FlowId -> [a] -> m ()
-delayedPublish flowId events  = do
+delayedPublish maybeFlowId events  = do
   liftIO $ threadDelay (10^6)
-  eventsPublish myEventTypeName flowId events
+  let flowId = fromMaybe (FlowId "shalom") maybeFlowId
```
Comment (etorreborre) [inline tests/Network/Nakadi/Tests/Common.hs]: I like the flow id name :-)

Comment (etorreborre) [reply tests/Network/Nakadi/Tests/Common.hs]: Is there a way to test that we really emit the `X-Flow-Id` header? Or maybe add a unit test for `includeFlowId`?

**Comment by etorreborre** [comment]
👍

File: src/Network/Nakadi/EventTypes/EventType.hs (hunk 1 above)
Comment (mtesseract) [reply src/Network/Nakadi/EventTypes/EventType.hs]: True. I don't have strong feelings in this regard, but I have now moved the includeFlowId call to the top (before setRequestpath).

File: tests/Network/Nakadi/Tests/Common.hs (hunk 2 above)
Comment (mtesseract) [reply tests/Network/Nakadi/Tests/Common.hs]: I have added tests.

**Comment by etorreborre** [comment]
Time for the release or do you have anything else in mind?

**Comment by mtesseract** [comment]
I hope I can do a release tonight.

**Comment by mtesseract** [comment]
Thanks!

**Pull request merged**